test/distributed/test_device_mesh.py

```diff
@@ -320,6 +322,44 @@ def test_get_item_1d(self):
         with self.assertRaisesRegex(RuntimeError, "Invalid mesh_dim_name"):
             dp_mesh = mesh["dim0"]
 
+    def _get_tags_to_pg_name(self, tags_to_pg):
+        tags_to_pg_name = {}
+        for tag_name, groups in tags_to_pg.items():
+            tags_to_pg_name[tag_name] = []
+            for group in groups:
+                tags_to_pg_name[tag_name].append(_get_process_group_name(group))
+        return tags_to_pg_name
+
+    @with_comms
+    @run_with_both_funcol_impls
+    def test_cache_and_reuse_submesh_slice_result(self):
+        mesh = init_device_mesh(self.device_type, (2, 4), mesh_dim_names=("dp", "tp"))
+
+        dp_mesh = mesh["dp"]
+        # Record the pg tags_to_pg_name so we can check whether the next mesh["dp"] slice create new pg.
+        # We cannot do a deepcopy of ProcessGroup for comparison,
+        # since we cannot pickle 'torch._C._distributed_c10d.ProcessGroup' object.
+        # Therefore, we rely on the tags_to_pg_name for comparison to make sure
+        # before/after two mesh["dp"] call, tags_to_pg in the current world does not change.
+        ref_tags_to_pg_name = self._get_tags_to_pg_name(_world.tags_to_pg)
```

Comment:
can we just compare the pg_dim directly, instead of trying to find the tag name then map it to pg_name? iirc the tag name would go away later too, so we should not rely on that